python/TestHarness/testers/SignalTester.py

```diff
@@ -0,0 +1,30 @@
+#* This file is part of the MOOSE framework
+#* https://www.mooseframework.org
+#*
+#* All rights reserved, see COPYRIGHT for full restrictions
+#* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#*
+#* Licensed under LGPL 2.1, please see LICENSE for details
+#* https://www.gnu.org/licenses/lgpl-2.1.html
+
+from RunApp import RunApp
+import os, signal,time
+
+# Classes that derive from this class are expected to write
+# output files. The Tester::getOutputFiles() method should
+# be implemented for all derived classes.
+class SignalTester(RunApp):
+    @staticmethod
+    def validParams():
+        params = RunApp.validParams()
+        params.addParam('signal', "SIGUSR1", "The signal to send to the app. Defaults to SIGUSR1")
+        params.addParam('sleep_time', 3, "The amount of time the tester should wait before sending a signal.")
+        return params
+
+    def __init__(self, name, params):
+        RunApp.__init__(self, name, params)
+
+    def check_signal(self,pid):
+        #wait for 3 seconds for the output to start, then send the SIGUSR1 signal to the program
```

Comment:
```suggestion
        # Wait for the number of seconds defined by sleep_time for the output to start, then send the SIGUSR1 signal to the program
```
I would adjust this comment a bit just so that its clear that 3 is only a default - the value of `sleep_time` is the only thing that matters.

Reply:
oh yeah this is an artifact of when sleep time wasn't a param, and it just always waited 3 sec